library/src/main/java/com/novoda/downloadmanager/DownloadNotificationChannelCreator.java

```diff
@@ -0,0 +1,36 @@
+package com.novoda.downloadmanager;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+
+class DownloadNotificationChannelCreator implements NotificationChannelCreator {
+
+    private final NotificationManager notificationManager;
+
+    DownloadNotificationChannelCreator(NotificationManager notificationManager) {
+        this.notificationManager = notificationManager;
+    }
+
+    private static String CHANNEL_ID;
+
+    @Override
+    public String createDownloadNotificationChannel(Context context) {
+        if (CHANNEL_ID == null) {
+            CHANNEL_ID = "download-manager";
+            createNotificationChannelForAndroidOreo(context);
+        }
+
+        return CHANNEL_ID;
+    }
+
+    private void createNotificationChannelForAndroidOreo(Context context) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            String channelName = "Download Manager Notification Service";
```

Comment:
is the plan to have this hardcoded? no resources?

Reply:
Good point @takecare, I'll add this to `resources`